styles/main/_form.scss

```diff
@@ -0,0 +1,373 @@
+// A HTML `<form>` is expected to consists of a sequence of one ore more
+// `<div>` elements.
+// Each of this `<div>` binds together a label, an input element and some
+// explanatory text.
+// This kind of `<div>` is called an input group.
+// There are several classes of input groups which define how their children
+// are arranged.
+//
+// First, this file defines styles for form elements which always apply
+// and which are independent of the class of the input group.
+// Second, this file defines the different input groups and adjusts some
+// form elements acc. to the class of input group.
+
+// 1. Form elements (general styling, independent of input group)
+
+input,
+div.select,
+select,
+textarea,
+output {
+	display: inline-block;
+	/* we must position this element so that it becomes the nearest
+   * positioned ancestor for ::before, ::after and :checked::before */
```

Comment:
```suggestion
	/* we must position this element so that it becomes the nearest
	 * positioned ancestor for ::before, ::after and :checked::before */
```